lib/Installation/Partitioner/LibstorageNG/FileSystemOptionsPage.pm

```diff
@@ -28,4 +29,11 @@ sub press_next {
     $self->get_navigation_panel()->press_next();
 }
 
+sub set_separate_home_partition_checkbox {
+    my ($self, $has_separate_home)     = @_;
+    assert_screen(FILE_SYSTEM_OPTIONS_PAGE);
+    my $separate_home_checkbox = Installation::Partitioner::SeparateHomeCheckbox->new('alt-r');
```

Comment:
I noticed here something but I could not check the VR (are not accessible this morning) so I maybe wrong. For example I see that you group all `new()` needed for the object in other cases, but in this case you are doing on-the-fly. I think is ok if there is some condition or something special, like is not visible on the screen on that moment, but in general could it have more sense to have all objects created inside the object that is created in initialization time? WDYT?

Reply:
I also thought about this solution and maybe it is the right way. If two of us already thought in the same direction, than probably it is better. Thanks:)